### PR TITLE
Fix/confirm outgoing amount currency

### PR DESCRIPTION
**Pull request by stevenzeiler**
Using the source_balance_changes and destination_balance_changes the outgoing payment record is updated with what actually happened.

### CODE REVIEW

File: test/core/outgoing_payment.js [hunk 1]
```diff
@@ -159,20 +159,18 @@ describe('Sending a queued payment to ripple', function() {
...
           .then(function(status) {
             chai.assert.strictEqual(outgoingPayment.record.state, 'failed');
             done();
-          }).error(function(error){
-            throw new Error(error);
-          });
+          })
       })
       .error(function(error){
         throw new Error(error);
```
Comment (cdmcnamara) [inline test/core/outgoing_payment.js]: You can do 
```
done(new Error(error));
```

File: test/core/outgoing_payment.js [hunk 2]
```diff
@@ -257,4 +255,34 @@ describe('Sending a queued payment to ripple', function() {
...
+      .then(function(rippleTransaction){
+        outgoingPayment = new OutgoingPayment(rippleTransaction);
+        return outgoingPayment._recordAcceptanceOrRejectionStatus(validatedPayment)
+      })
+      .then(function(rippleTransaction) {
+        chai.assert.strictEqual(rippleTransaction.to_currency, 'XAU');
+        chai.assert.strictEqual(rippleTransaction.to_amount, 0.7);
+        chai.assert.strictEqual(rippleTransaction.from_currency, 'BTC');
+        chai.assert.strictEqual(rippleTransaction.from_amount, 1.45);
+        done();
```
Comment (cdmcnamara) [inline test/core/outgoing_payment.js]: Might want to add:
```
.error(function(error) {
  done(new Error(error);
})
```
Just to properly catch anything issues

Comment (stevenzeiler) [inline test/core/outgoing_payment.js]: Okay I will add an error handler.

**Comment by cdmcnamara** [comment]
LGTM.

**Comment by stevenzeiler** [comment]
Thanks, I will merge.

**Pull request merged**